### PR TITLE
build: extra require for `langchain`, `minichain`, `transformers`

### PR DESCRIPTION
**Pull request by bdura**
## Description

This PR proposes three extras:
- `langchain`
- `minichain`
- `hf`, for running Hugging Face models

### Types of change

This is a change in the dependencies.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran all tests , and all new and existing tests passed. This includes tests marked as `external` and those 
  requiring a GPU. I did this for the following directories:
  - [x] `tests`
  - [x] `usage_example/tests`
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -39,6 +39,16 @@ install_requires =
...
 
+[options.extras_require]
+minichain =
+    minichain>=0.3,<0.4
+langchain =
+    langchain>=0.0.144,<0.1
+hf =
+    cupy-cuda11x
+    torch>=1.13.1,<2.0
+    transformers>=4.28.1,<5.0
```
Comment (adrianeboyd) [inline setup.cfg:50]: To be parallel to the spacy extras, I would call this option `transformers` and leave out `cupy`. (I don't think you need it and if you do, `cupy-cuda11x` is too specific as a version. I can make suggestions if you do need to keep it here.)

Comment (bdura) [reply setup.cfg]: Good point. I aligned the packages on the README, maybe there's something to change there as well.

File: README.md [hunk 2]
```diff
@@ -514,6 +518,8 @@ This allows you to have the setting `device=cuda:0` in your config, which ensure
 python -m pip install "cupy-cuda11x"
 python -m pip install "torch>=1.13.1,<2.0"
 python -m pip install "transformers>=4.28.1,<5.0"
+# Or install with spacy-llm directly
+python -m pip install "spacy-llm[hf]"
```
Comment (adrianeboyd) [inline README.md:522]:
```suggestion
python -m pip install "spacy-llm[transformers]"
```

**Comment by adrianeboyd** [comment]
You need cupy for thinc on GPU, but for transformers it only depends on your torch install.

**Review by rmitsch — COMMENTED** [review]
Looks good!

> You need cupy for thinc on GPU, but for transformers it only depends on your torch install.

@adrianeboyd How should we reflect this in `setup.cfg`?

**Comment by adrianeboyd** [comment]
For `spacy-llm` I don't think you need to worry about cupy in general?

Very probably people will want more control over the torch devices, which we'd usually do through `require_gpu` and `require_cpu` with spacy in general, but I would not worry about this for this PR.

**Pull request merged**